primitives/src/witness.rs

```diff
@@ -386,6 +386,16 @@ impl<'a> IntoIterator for &'a Witness {
     fn into_iter(self) -> Self::IntoIter { self.iter() }
 }
 
+impl<T: AsRef<[u8]>> FromIterator<T> for Witness {
```

Comment:
In 4ae34c4e18c2224e30168665f1e45569f30c4967:

Also, @tcharding just confirming -- do we actually want this for every `T: AsRef<[u8]>` type? Notably this includes `str`, which means that somebody will be able to call `.collect` on a list of strings and the strings will just turn into ASCII witness elements, which is inconsistent with #4366 (and prevents us from later doing some hex-decoding version of this).

I *think* this is fine, and just another symptom of Rust thinking that hex encoding is "niche" and forcing every developer to make bad API decisions forever, but I want to highlight it.

Reply:
I think it's fine. It doesn't prevent us from doing hex decoding because it's possible to use `.map(hex::decode_to_vec)` and converting strings without mentioning "hex" anywhere sounds too implicit anyway.

If anything, I'd say it's questionable that `std` added `AsRef<[u8]>` for strings.